src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1792,7 +1792,12 @@ public function getRouteKeyName()
      */
     public function resolveRouteBinding($value, $field = null)
     {
-        return $this->where($field ?? $this->getRouteKeyName(), $value)->first();
+        try {
+            return $this->where($field ?? $this->getRouteKeyName(), $value)->first();
+        } catch (QueryException $e) {
+            // URL type failed to cast into the DB type
+            return null;
+        }
```

Comment:
```suggestion
        rescue(function () use ($field, value) {
            return $this->where($field ?? $this->getRouteKeyName(), $value)->first();
        }, null);
```

This should be sufficient.

Reply:
@crynobone don't you mean to return the value from `rescue`?

Reply:
Sorry, missed that out

Reply:
@fcolecumberri Can you update the pr with Mior's proposal?

Reply:
Done